rust/tableau_summary/src/twb/diff/util.rs

```diff
@@ -341,21 +478,120 @@ mod tests {
             vec!["bar", "baz"],
             vec!["foo", "bar", "baz"],
             vec![
-                (ChangeState::Change, Some("bar"), Some("foo")),
-                (ChangeState::Change, Some("baz"), Some("bar")),
-                (ChangeState::Add, None, Some("baz")),
+                (ChangeState::Add, None, Some("foo")),
+                (ChangeState::None, Some("bar"), None),
+                (ChangeState::None, Some("baz"), None),
```

Comment:
This is an example of the simplification this PR makes:
when prepending "foo" to the list, instead of that being treated as "2 changes and an add", it is just an "add and two no-ops".